src/main/java/internet/store/dao/jdbc/ShopCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,141 @@
+package internet.store.dao.jdbc;
+
+import internet.store.dao.ShopCartDao;
+import internet.store.exception.DataProcessingException;
+import internet.store.lib.Dao;
+import internet.store.model.Product;
+import internet.store.model.ShoppingCart;
+import internet.store.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShopCartDaoJdbcImpl implements ShopCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        String query = "SELECT user_id, cart_id, id, product_name, price FROM users "
+                + "JOIN shopping_cart using(user_id) "
+                + "LEFT JOIN shopping_cart_product USING(cart_id) "
+                + "LEFT JOIN products ON products.id = shopping_cart_product.product_id "
+                + "WHERE user_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            ShoppingCart shoppingCart = new ShoppingCart(userId);
+            List<Product> productList = new ArrayList<>();
+            if (resultSet.next()) {
+                shoppingCart.setCartId(resultSet.getLong("cart_id"));
+            }
+            do {
+                Long productId = resultSet.getObject("id", Long.class);
+                if (productId != null) {
+                    productList.add(retrieveProductDataFromDB(resultSet));
+                }
+            } while (resultSet.next());
```

Comment:
let's move this logic to separate method

Reply:
We discussed this with Sofia. I couldn't work around of this yesterday and she suggested to see mentor's comments ;) - so I expect this. Will think about it and find solution...